File: 2024/day01/part1.rs

use std::fs;
use std::vec::Vec;

fn main(){
    const FILE_PATH:&str = "./in.txt";
    let content = fs::read_to_string(FILE_PATH).expect("Should have been able to read the file");
    let mut v1 : Vec<i64> = Vec::new();
    let mut v2 : Vec<i64> = Vec::new();
    for line in content.lines(){
        let parts: Vec<i64> = line
        .split_whitespace()
        .map(|num| num.parse::<i64>().unwrap())
        .collect();
        v1.push(parts[0]);
        v2.push(parts[1]);
    }
    v1.sort_unstable();
    v2.sort_unstable();
    let mut ans:u64 = 0;
    for i in 0..v1.len(){
        ans+=v1[i].abs_diff(v2[i]);
    }
    println!("{}",ans);
}

File: 2024/day13/part1.py
import re
import math
f = open("in.txt","r").read().split("\n\n")
ans=0
for i in f:
    A,B,P=i.split('\n')
    AX,AY=map(int,re.findall("\d+",A))
    BX,BY=map(int,re.findall("\d+",B))
    PX,PY=map(int,re.findall("\d+",P))
    x=(PX*BY-PY*BX)/(AX*BY-AY*BX)
    y=(PX-AX*x)/BX
    if x%1==0 and y%1==0:
        ans+=int(3*x+y)
print(ans)

